src/tokens/tokens.utils.ts

```diff
@@ -115,6 +120,78 @@ export async function getCachedTokenBalances(
   return undefined;
 }
 
+export async function getFullToken(chain: Chain, tokenAddr: Token['address']): Promise<TokenFull | null> {
+  const fullTokenMap = await getFullTokens(chain, [tokenAddr]);
+  return fullTokenMap[tokenAddr] || null;
+}
+
+export async function getFullTokens(chain: Chain, tokensAddr: Token['address'][]): Promise<TokenFullMap> {
+  const cachedTokens = await getCachedTokesInfo(tokensAddr);
+  const cachedTokensAddr = cachedTokens.map((token) => token.address);
```

Comment:
should this be a set?

Reply:
ye, i thought about that, but u cant subtracte Sets in js :(

Reply:
ye.. but no difference with filter at all